app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    previous_runs = {}
+
+    def inner(*args) -> object:
+        nonlocal previous_runs
+        hash_key = str(func.__name__) + str(args)
```

Comment:
```suggestion
        hash_key = args
```
No need to add a function name to the key because the cache dict is unique for each function. It is created while decorating.

Reply:
Changes were accepted.